apps/wallet-mobile/src/navigation.tsx

```diff
@@ -514,6 +542,28 @@ export const useWalletNavigation = () => {
   } as const).current
 }
 
+export const useNavigateToWalletSelectionSync = () => {
+  const {resetToWalletSetup, resetToWalletSelection} = useWalletNavigation()
+  const walletManager = useWalletManager()
+  const {refetch} = useHasWallets(walletManager, {
+    enabled: false,
+    onSuccess: (hasWallets) => {
+      if (hasWallets) {
+        resetToWalletSelection()
+        return
+      }
+
+      resetToWalletSetup()
+    },
+  })
+
+  const navigateToWalletSelectionSync = React.useCallback(() => {
+    refetch()
+  }, [refetch])
+
+  return navigateToWalletSelectionSync
+}
+
```

Comment:
To determine if the user has wallets, it is necessary to use the `listWallets` method.

When deleting a wallet, its metadata is not deleted at that time, but rather when the app is launched. When using the `listWallets` method, a check is performed to filter the IDs of the wallets that are going to be deleted.

In short, it is necessary to perform a `refetch` to obtain an updated value of `hasWallets`, that is needed to determine if we are going to show the wallets list screen or the setup wallet screen